### PR TITLE
bs - add location to controller for course history

### PR DESCRIPTION
**Pull request by BrennaScholte**
In this pull request I added location as a parameter in the course history controller. The goal is to eventually implement a location search option in course history.

Closes: #37

### CODE REVIEW

**Review by kkwen123 — APPROVED** [review]
LGTM

**Review by Joshuason55 — APPROVED** [review]
Everything looks good!

File: src/main/java/edu/ucsb/cs156/courses/controllers/CourseOverTimeController.java [hunk 1]
```diff
@@ -35,7 +35,8 @@ public ResponseEntity<String> search(
         @RequestParam String startQtr,
         @RequestParam String endQtr,
         @RequestParam String subjectArea,
-        @RequestParam String courseNumber
+        @RequestParam String courseNumber,
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/controllers/CourseOverTimeController.java:38]: I'm not sure I understand why we are adding this parameter to the controller endpoint, when that parameter is not being passed to the query.

The course results use the other four parameters, but not this one.  So this seems misleading.

Can you help me understand your intention here?

The code that actually gets this courses is this code from a few lines down, and it doesn't do anything with the `location` parameters.

```java
courseResults = convertedSectionCollection.findByQuarterRangeAndCourseId(
            startQtr,
            endQtr,
            makeFormattedCourseId(subjectArea, courseNumber)
);
```

**Review by pconrad — CHANGES_REQUESTED** [review]
I'm confused... the addition of this parameter seems to do nothing.

What is this PR intended to accomplish?

It seems as if you intend to be able to search  by location.

But if you want to do that, you need to start by adding a method to the `ConvertedSectionCollection` that will support what you are trying to do.

It would also probably make more sense, and be less risky to create a new controller endpoint for searching by location, rather than combining that with the search by instructor.

By combining them, you create the need to test what happens in four circumstances:
* courseNumber blank, location blank
* courseNumber blank, location specified
* courseNumber specified, location blank
* courseNumber specified, location specified

And all four have to do something reasonable.

If you make a new api endpoint instead, you can make location required, and now you have only one endpoint to worry about, and you don't have to worry about breaking an existing one.

File: src/main/java/edu/ucsb/cs156/courses/controllers/CourseOverTimeController.java (hunk 1 above)
Comment (BrennaScholte) [reply src/main/java/edu/ucsb/cs156/courses/controllers/CourseOverTimeController.java:38]: Ok my apologizes. I didn't realize how much this change would impact everything. I will go back and rethink how to approach this issue.

**Pull request closed without merging**